lib/publiccloud/basetest.pm

```diff
@@ -113,6 +113,7 @@ sub _cleanup {
 
     eval { $self->cleanup(); } or bmwqemu::fctwarn("self::cleanup() failed -- $@");
 
+    return if !($self->{result});    # skip when result is undef
```

Comment:
I vote for dropping this line with `publiccloud_multi_module` all should be fine without it

Reply:
I agree that bigger refactoring of this logic is needed and I agree that current approach is messy but this "bigger refactoring" won't contain this line either ...